orangecontrib/text/widgets/owdocumentembedding.py

```diff
@@ -0,0 +1,236 @@
+        unsuccessful_embeddings = Msg('Some embeddings were unsuccessful.')
+
+    
+    language = Setting(default = 'English')
+    aggregator = Setting(default = 'mean')
+
+    def __init__(self):
+        OWWidget.__init__(self)
+        ConcurrentWidgetMixin.__init__(self)
+
+        self.languages = list(LANGS_TO_ISO.keys())
+        self.aggregators = AGGREGATORS
+        self.corpus = None
+        self.new_corpus = None
+        self._setup_layout()
+
+    def _setup_layout(self):
+        self.controlArea.setMinimumWidth(self.controlArea.sizeHint().width())
+
+        widget_box = widgetBox(self.controlArea, 'Settings')
+
+        self.language_cb = comboBox(widget = widget_box,
+                                    master = self,
+                                    value = 'language',
+                                    label = 'Language: ',
+                                    orientation = Qt.Horizontal,
+                                    items = self.languages,
+                                    callback = self._option_changed
+        )
+
+        self.aggregator_cb = comboBox(widget = widget_box,
+                                      master = self,
+                                      value = 'aggregator',
+                                      label = 'Aggregator: ',
+                                      orientation = Qt.Horizontal,
+                                      items = self.aggregators,
+                                      callback = self._option_changed 
+        )
+
+        self.auto_commit_widget = auto_commit(widget = self.controlArea,
+                                              master = self,
+                                              value = '_auto_apply',
+                                              label = 'Apply',
+                                              commit = self.commit)
+
+        self.cancel_button = QPushButton(
+                                    'Cancel',
+                                    icon = self.style()
+                                    .standardIcon(QStyle.SP_DialogCancelButton))
+
+        self.cancel_button.clicked.connect(self.cancel)
+
+        hbox = hBox(self.controlArea)
+        hbox.layout().addWidget(self.cancel_button)
+        self.cancel_button.setDisabled(True)
+
+    def set_input_corpus_summary(self, corpus):
+        if(corpus is None):
+            self.info.set_input_summary(self.info.NoInput)
+        else:
+            self.info.set_input_summary("{} documents."
+                                        .format(len(corpus)))
+
+    def set_output_corpus_summary(self, corpus):
+        if(corpus is None):
+            self.info.set_output_summary(self.info.NoOutput)
+        else:
+            unsuccessful = np.sum(np.isnan(corpus.X)) / EMB_DIM
+            if(unsuccessful > 0):
+                self.Warning.unsuccessful_embeddings()
+            self.info.set_output_summary("Successful: {}, Unsuccessful: {}"
```

Comment:
Same than for input: first argument should be shorter. All this string go to second.